src/docfx/lib/markdown/HtmlExtension.cs

```diff
@@ -54,7 +54,7 @@ string ProcessHtml(string html, MarkdownObject block, ErrorBuilder errors, bool
                 // The following code ensures we preserve the original html when changing links.
                 return HtmlUtility.TransformHtml(html, (ref HtmlReader reader, ref HtmlWriter writer, ref HtmlToken token) =>
                 {
-                    HtmlUtility.TransformLink(ref token, block, getLink, getImageLink!);
+                    HtmlUtility.TransformLink(ref token, block, getLink!, getImageLink!);
```

Comment:
remove `!` ?
```suggestion
                    HtmlUtility.TransformLink(ref token, block, getLink, getImageLink);
```

Reply:
that will result in lots of code warning:

Severity	Code	Description	Project	File	Line	Suppression State
Warning	CS8620	Argument of type 'Func<SourceInfo<string>, MarkdownObject, string>' cannot be used for parameter 'transformLink' of type 'Func<SourceInfo<string>, MarkdownObject?, string>' in 'void HtmlUtility.TransformLink(ref HtmlToken token, MarkdownObject? block, Func<SourceInfo<string>, MarkdownObject?, string> transformLink, Func<SourceInfo<string>, MarkdownObject?, string?, int, string>? transformImageLink = null)' due to differences in the nullability of reference types.	docfx	C:\Users\fan\source\repos\docfx\src\docfx\lib\markdown\HtmlExtension.cs	57	Active

Reply:
Maybe we can not remove it now, and this exclamation mark is an exception according to our original talking. : ) @yufeih